client/Components/createEggModal.jsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import Button from '../UI/button';
+
+class CreateEgg extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      message: '',
+      user: this.props.user.username,
+      userId: this.props.user.id
+    };
+    this.fileInputRef = React.createRef();
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleMessageChange = this.handleMessageChange.bind(this);
+  }
+
+  handleMessageChange(event) {
+    this.setState({ message: event.target.value });
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    const formData = new FormData();
+    formData.append('message', this.state.message);
+    formData.append('image', this.fileInputRef.current.files[0]);
+    formData.append('longitude', this.props.eggLocation.longitude);
+    formData.append('latitude', this.props.eggLocation.latitude);
+    formData.append('user', this.state.userId);
```

Comment:
Instead of putting the userId in the request body, add a token to your request headers and have the server verify that the request is on behalf of that specific user.